Worker docker builds are broken

```
 => ERROR [ecr-login-installation 3/5] RUN go get -u github.com/awslabs/amazon-ecr-credential-helper/ecr-  5.3s
------
 > [ecr-login-installation 3/5] RUN go get -u github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login:
#0 2.710 # github.com/awslabs/amazon-ecr-credential-helper/ecr-login/vendor/golang.org/x/sys/unix
#0 2.710 /root/go/src/github.com/awslabs/amazon-ecr-credential-helper/ecr-login/vendor/golang.org/x/sys/unix/syscall.go:83:16: undefined: unsafe.Slice
#0 2.710 /root/go/src/github.com/awslabs/amazon-ecr-credential-helper/ecr-login/vendor/golang.org/x/sys/unix/syscall_linux.go:1018:20: undefined: unsafe.Slice
#0 2.710 /root/go/src/github.com/awslabs/amazon-ecr-credential-helper/ecr-login/vendor/golang.org/x/sys/unix/syscall_linux.go:2297:9: undefined: unsafe.Slice
#0 2.710 /root/go/src/github.com/awslabs/amazon-ecr-credential-helper/ecr-login/vendor/golang.org/x/sys/unix/syscall_unix.go:118:7: undefined: unsafe.Slice
#0 2.710 /root/go/src/github.com/awslabs/amazon-ecr-credential-helper/ecr-login/vendor/golang.org/x/sys/unix/sysvshm_unix.go:33:7: undefined: unsafe.Slice
------
Dockerfile.worker:18
--------------------
  16 |         make -C builddir install;
  17 |     
  18 | >>> RUN go get -u github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login
  19 |     WORKDIR /root/go/src/github.com/awslabs/amazon-ecr-credential-helper
  20 |     RUN make
--------------------
ERROR: failed to solve: process "/bin/sh -c go get -u github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login" did not complete successfully: exit code: 2
Traceback (most recent call last):
  File "/home/ubuntu/environment/codalab/codalab-worksheets/venv/bin/codalab-service", line 8, in <module>
    sys.exit(main())
  File "/home/ubuntu/environment/codalab/codalab-worksheets/codalab_service.py", line 125, in main
    service_manager.execute()
  File "/home/ubuntu/environment/codalab/codalab-worksheets/codalab_service.py", line 784, in execute
    self.build_images()
  File "/home/ubuntu/environment/codalab/codalab-worksheets/codalab_service.py", line 1049, in build_images
    self.build_image(image)
  File "/home/ubuntu/environment/codalab/codalab-worksheets/codalab_service.py", line 831, in build_image
    self._run_docker_cmd(
  File "/home/ubuntu/environment/codalab/codalab-worksheets/codalab_service.py", line 855, in _run_docker_cmd
    raise e
  File "/home/ubuntu/environment/codalab/codalab-worksheets/codalab_service.py", line 850, in _run_docker_cmd
    subprocess.check_call(command_string, shell=True, cwd=BASE_DIR)
  File "/home/ubuntu/.pyenv/versions/3.8.10/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'docker build  -t codalab/worker:rc1.7.0 -f docker_config/dockerfiles/Dockerfile.worker .' returned non-zero exit status 1.
```

--- a/docker_config/dockerfiles/Dockerfile.worker
+++ b/docker_config/dockerfiles/Dockerfile.worker
@@ -1,22 +1,3 @@
-FROM golang:1.17 as ecr-login-installation
-# Install the Amazon ECR Docker Credential Helper
-# Use Docker multi-stage builds to get rid of intermediate layers related to installation
-
-ENV GOPATH /root/go
-ENV SINGULARITY_VERSION=3.7.4
-
-RUN mkdir -p /usr/local/var/singularity/mnt && \
-    mkdir -p $GOPATH/src/github.com/sylabs && \
-    cd $GOPATH/src/github.com/sylabs && \
-    wget https://github.com/sylabs/singularity/releases/download/v${SINGULARITY_VERSION}/singularity-${SINGULARITY_VERSION}.tar.gz && \
-    tar -xzvf singularity-${SINGULARITY_VERSION}.tar.gz && \
-    cd singularity && \
-    ./mconfig -p /usr/local && \
-    make -C builddir && \
-    make -C builddir install;
-
-RUN go install github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login@latest
-
 FROM ubuntu:20.04
 MAINTAINER CodaLab Worksheets <codalab.worksheets@gmail.com>
 ENV DEBIAN_FRONTEND noninteractive
@@ -25,6 +6,7 @@ RUN apt-get update && \
   apt-get install -y --no-install-recommends software-properties-common curl && \
   apt-get update && \
   apt-get install -y --no-install-recommends \
+    amazon-ecr-credential-helper \
     build-essential \
     libssl-dev \
     uuid-dev \
@@ -45,11 +27,6 @@ RUN curl -o ~/miniconda.sh -O  https://repo.anaconda.com/miniconda/Miniconda3-4.
      rm ~/miniconda.sh
 ENV PATH /opt/conda/bin:$PATH
 RUN conda --version
-
-COPY --from=ecr-login-installation /root/go/src/github.com/awslabs/amazon-ecr-credential-helper/bin/local /usr/local/bin
-COPY --from=ecr-login-installation /usr/local/bin/singularity /usr/local/bin
-# the singularity container runtime needs to be able to lookup the uid of the default user
-#RUN useradd -ms /bin/bash $default_user
 
 RUN mkdir $HOME/.docker
 RUN echo "{\"credsStore\": \"ecr-login\"}" >> ~/.docker/config.json

--- a/docker_config/dockerfiles/Dockerfile.worker
+++ b/docker_config/dockerfiles/Dockerfile.worker
@@ -1,4 +1,4 @@
-FROM golang:1.15 as ecr-login-installation
+FROM golang:1.17 as ecr-login-installation
 # Install the Amazon ECR Docker Credential Helper
 # Use Docker multi-stage builds to get rid of intermediate layers related to installation
 

--- a/docker_config/dockerfiles/Dockerfile.worker
+++ b/docker_config/dockerfiles/Dockerfile.worker
@@ -15,7 +15,7 @@ RUN mkdir -p /usr/local/var/singularity/mnt && \
     make -C builddir && \
     make -C builddir install;
 
-RUN go get -u github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login
+RUN go install github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login@latest
 WORKDIR /root/go/src/github.com/awslabs/amazon-ecr-credential-helper
 RUN make
 

--- a/docker_config/dockerfiles/Dockerfile.worker
+++ b/docker_config/dockerfiles/Dockerfile.worker
@@ -16,8 +16,6 @@ RUN mkdir -p /usr/local/var/singularity/mnt && \
     make -C builddir install;
 
 RUN go install github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login@latest
-WORKDIR /root/go/src/github.com/awslabs/amazon-ecr-credential-helper
-RUN make
 
 FROM ubuntu:20.04
 MAINTAINER CodaLab Worksheets <codalab.worksheets@gmail.com>

